orangecontrib/snom/preprocess/utils.py

```diff
@@ -148,6 +182,17 @@ def domain_with_single_attribute_in_x(attribute, domain):
     metas = [a for a in domain.metas if a.name != attribute.name]
     return Domain([attribute], class_vars, metas)
 
+def table_with_no_attribute(data):
+    """Create a domain with only the attribute in domain.attributes and ensure
+    that the same attribute is removed from metas and class_vars if it was present
+    there."""
+    class_vars = [a for a in data.domain.class_vars]
+    metas = [a for a in data.domain.metas]
+
+    newdomain = Domain([], class_vars, metas)
```

Comment:
Here you could just use `newdomain = Domain([], data.domain.class_vars, data.domain.metas)` (no need for loops)